pkg/dao/task_info.go

```diff
@@ -20,11 +36,16 @@ func GetTaskInfoDao(db *gorm.DB) TaskInfoDao {
 	}
 }
 
-func (t taskInfoDaoImpl) Fetch(orgId string, id string) (api.TaskInfoResponse, error) {
-	taskInfo := models.TaskInfo{}
-	result := t.db.Where("id = ? AND org_id = ?", id, orgId).First(&taskInfo)
+func (t taskInfoDaoImpl) Fetch(orgID string, id string) (api.TaskInfoResponse, error) {
+	taskInfo := models.TaskInfoRepositoryConfiguration{}
 	taskInfoResponse := api.TaskInfoResponse{}
 
+	result := t.db.Table(taskInfo.TableName()+" AS t ").
+		Select(JoinSelectQuery).
+		Joins("INNER JOIN repositories r on t.repository_uuid = r.uuid").
+		Joins("INNER JOIN repository_configurations rc on r.uuid = rc.repository_uuid").
+		Where("t.id = ? AND t.org_id = ?", id, orgID).First(&taskInfo)
```

Comment:
by doing an inner join, you exclude any tasks that don't have an associated repository or repository_configuration.  We probably want to do a Left join, to include those even if the resulting select returns a null repo_config_uuid/name.